code-challenges.js

```diff
@@ -41,18 +91,65 @@ var vowelTester1 = "learn"
 var vowelTester2 = "throw"
 // Expected output: 3
 
+//declare a function that takes in a string
+const firstVowelIndex = (string) => {
+    //create a new variable, split the string into an array, and use map to get the indexes of where the vowels are in the word.
+    let getIndex = string.split("").map((value, index) => {
+        if (value === "a" || value === "e" || value === "i" || value === "o" || value === "u"){
+            //return the indexes
+            return index
+        }
+    })
+    //filter out the undefined values
+    let noUndefined = getIndex.filter(value => value !== undefined)
+    //return the first index of the array of indexes, which is the index of the first vowel
+    return noUndefined[0]
+}
 
+// console.log(firstVowelIndex(vowelTester1));
+// console.log(firstVowelIndex(vowelTester2));
 
 
 
 // --------------------5) Create a function that takes the toonimals array and returns an array with only the toon objects that are cats.
 // Expected output: [ { name: "Stimpy", animal: "cat" }, { name: "Scratchy", animal: "cat" }, { name: "Felix", animal: "cat" } ]
 
-var toonimals = [ {name: "Itchy", animal: "mouse"}, {name: "Stimpy", animal: "cat"}, {name: "Daffy", animal: "duck"}, {name: "Scratchy", animal: "cat"}, {name: "Ren", animal: "dog"}, {name: "Felix", animal: "cat"}]
+var toonimals = [  
+    {name: "Itchy", animal: "mouse"}, 
+    {name: "Stimpy", animal: "cat"}, 
+    {name: "Daffy", animal: "duck"}, 
+    {name: "Scratchy", animal: "cat"}, 
+    {name: "Ren", animal: "dog"}, 
+    {name: "Felix", animal: "cat"}]
+
+//create a function that takes in an array
+const getCats = (array) => {
+    //create a new variable that filters out any animal that is a cat
+    var animalCat = array.filter(value => 
+        value.animal === "cat")
+    //return the names and animal types of just cats
+    return animalCat
+}
 
+  
 
+// console.log(getCats(toonimals));
 
 
 
 // --------------------6) Using the toonimals variable, create a function that returns only the names of the non-cats.
 // Expected output: "Itchy" "Daffy" "Ren"
+
+//create a function that takes in an array
+const onlyNonCatNames = (array) => {
+    //create a variable that uses filter to return any animal that is not a cat
+    var noCats = array.filter(value => value.animal !== "cat")
+    //use map to go through the previous array created (noCats) and only return the names of the non-cats.
+    return noCats.map(value => {
+        return value.name
+    })
+    }
+
+
+
+// console.log(onlyNonCatNames(toonimals));
```

Comment:
Looking good! This is just a bit nit picky, but since our expected output is doesn't include an array, but rather three separate strings, I would keep a look out for that!  For now, this is fine :)